src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -236,11 +261,34 @@ private async Task ChangeOrganizationSponsorship(Organization org, OrganizationS
             throw new GatewayException("Could not find customer payment profile.");
         }
 
-        if (customer.Address is null &&
+        var pm5766AutomaticTaxIsEnabled = _featureService.IsEnabled(FeatureFlagKeys.PM5766AutomaticTax);
+        var taxInfo = upgrade.TaxInfo;
+
+        if (!pm5766AutomaticTaxIsEnabled &&
+            taxInfo != null &&
+            !string.IsNullOrWhiteSpace(taxInfo.BillingAddressCountry) &&
+            !string.IsNullOrWhiteSpace(taxInfo.BillingAddressPostalCode))
+        {
+            var taxRateSearch = new TaxRate
+            {
+                Country = taxInfo.BillingAddressCountry,
+                PostalCode = taxInfo.BillingAddressPostalCode
+            };
+            var taxRates = await _taxRateRepository.GetByLocationAsync(taxRateSearch);
+
+            // should only be one tax rate per country/zip combo
+            var taxRate = taxRates.FirstOrDefault();
```

Comment:
⛏️ Seems like we're executing this block (272 - 280) several times. Might be worth extracting into a private method.